ipfs-proxy/src/index.js

```diff
@@ -0,0 +1,127 @@
+const Busboy = require('busboy')
+const imageType = require('image-type')
+const isJSON = require('is-json')
+const http = require('http')
+const httpProxy = require('http-proxy')
+const request = require('superagent')
+const zlib = require('zlib')
+
+const config = require('./config')
+const logger = require('./logger')
+
+const validImageTypes = ['image/jpeg', 'image/gif', 'image/png']
+
+function isValidFile(buffer) {
+  return isValidImage(buffer) || isJSON(buffer.toString())
+}
+
+function isValidImage(buffer) {
+  const image = imageType(buffer)
+  return image && validImageTypes.includes(image.mime)
+}
+
+function handleFileUpload (req, res) {
+  const busboy = new Busboy({
+    headers: req.headers,
+    limits: {
+      fileSize: 2 * 1024 * 1024
+    }
+  })
+
+  busboy.on('file', function(fieldname, file, filename, encoding, mimetype) {
+    file.fileRead = []
+
+    file.on('data', function(chunk) {
+      file.fileRead.push(chunk)
+    })
+
+    file.on('limit', function() {
+      logger.warn(`File upload too large`)
+      res.writeHead(413, { 'Connection': 'close' });
+      res.end()
+      req.unpipe(req.busboy)
+    })
+
+    file.on('end', function() {
+      let buffer = Buffer.concat(file.fileRead);
+
+      if (!isValidFile(buffer)) {
+        logger.warn(`Upload of invalid file type attempted`)
+        res.writeHead(415, { 'Connection': 'close' })
+        res.end()
+        req.unpipe(req.busboy)
+      } else {
+        const url = config.IPFS_API_URL + req.url + '?stream-channels=false'
+        request.post(url)
+          .set(req.headers)
+          .attach('file', buffer)
+          .then((response) => {
+            let responseData = response.text
+            if (response.headers['content-encoding'] == 'gzip') {
```

Comment:
Trying to understand if we need to support other types of compression...
Does the IPFS server obey to request encoding headers or does it always use gzip ?
For example if the browser sends in request that it supports compression of type "deflate". Does IPFS server still sends response compressed using gzip ?

Reply:
Good catch! I will add deflate support and test cases for both gzip and deflate.